### ISSUE TITLE
Block extra data

### ISSUE THREAD

**Issue by SwadicalRag** [glua/gm_voxelate#40]
This means that each block will occupy more than 2 bytes (when we implement this)

kind of like damage values to store extra data in blocks??? how do we implement this???

how do we implement chests, for example?

**Comment by SwadicalRag** [comment]
a separate std::unsigned_map linking an XYZCoordinate to a BlockBonusData struct or `std::tuple<size_t, void *>` to leave it up to the end user to figure out what they want????

**Comment by SwadicalRag** [comment]
I think a separate unordered map of coordinates and a std::string is a good idea. We could store the bonus data in json or bON from lua

**Comment by xaviergmail** [comment]
Why are y'all even bothering with gmod at this point - sorry

**Comment by MDFL64** [comment]
wtf, just store it lua-side

**Comment by MDFL64** [comment]
moreover, just use source's builtin entities or our own entities for this

**Comment by SwadicalRag** [comment]
I disagree with using source engine entities for block data storage: there are too many limitations

And storing it Lua side would be easy, I just wanted your opinion on c++ or Lua for this

**Comment by SwadicalRag** [comment]
actually, if map saving is done through c++, we might as well do bonus data in C++

**Comment by SwadicalRag** [comment]
Not too hard
